lua/examples/policy.lua

```diff
@@ -147,12 +145,79 @@ local function lookup_simple_policy(simple_policy, pkt_info)
 	return nil
 end
 
+-- The following defines the LPM policies for Grantor.
+
+-- This file only contains an example set of Bogons IPv4 lists
+-- downloaded from http://www.team-cymru.org/Services/Bogons/fullbogons-ipv4.txt
+local bogons_ipv4_file = "lua/examples/bogons-ipv4.txt"
+local lpm = lpmlib.new_lpm(1024, 256)
+
+-- This file only contains an example set of Bogons IPv6 lists
+-- downloaded from http://www.team-cymru.org/Services/Bogons/fullbogons-ipv6.txt
+local bogons_ipv6_file = "lua/examples/bogons-ipv6.txt"
+local lpm6 = lpmlib.new_lpm6(1024, 65536)
+
+for line in io.lines(bogons_ipv4_file) do
+	local ip_addr, prefix_len = lpmlib.str_to_prefix(line)
+	lpmlib.lpm_add(lpm, ip_addr, prefix_len, 253)
+end
+
+for line in io.lines(bogons_ipv6_file) do
+	local ip_addr, prefix_len = lpmlib.str_to_prefix6(line)
+	lpmlib.lpm6_add(lpm6, ip_addr, prefix_len, 253)
+end
+
+-- Function that looks up the lpm policy for the packet.
+local function lookup_lpm_policy(lpm_handler, pkt_info)
+
+	local dest_ip
+
+	if pkt_info.inner_ip_ver == policylib.c.IPV4 then
+		local ipv4_hdr = ffi.cast("struct ipv4_hdr *",
+			pkt_info.inner_l3_hdr)
+		local policy_id = lpmlib.lpm_lookup(lpm_handler,
+			ipv4_hdr.src_addr)
+		if policy_id < 0 then
+			return nil
+		end
+
+		return groups[policy_id]
+	elseif pkt_info.inner_ip_ver == policylib.c.IPV6 then
```

Comment:
Style: since the previous body of this `if..elseif..else..end` always returns, we can make this `elseif` an independent `if`. And the `else` clause just a `return nil`, no need for the `else`.